File: code/2025/03/28/模拟栈.py
m = int(input())
s = []
for i in range(m):
    p = input().split()
    if p[0] == "push":
        s.append(p[1])
    elif p[0] == "pop":
        if s:
            s.pop()
    elif p[0] == "empty":
        print("NO" if s else "YES")
    else:
        print(s[-1] if s else "empty")

File: code/2025/03/28/模拟栈.cpp
#include <bits/stdc++.h>

class Solution {
public:
	Solution() {}

	std::vector<int> s;
	void solve() {
		std::string p;
		std::cin >> p;
		if (p == "push") {
			int x;
			std::cin >> x;
			s.push_back(x);
		} else if (p == "pop") {
			if (not s.empty()) {
				s.pop_back();
			}
		} else if (p == "empty") {
			std::cout << (s.empty() ? "YES" : "NO") << "\n";
		} else {
			if (s.empty()) {
				std::cout << "empty";
			} else {
				std::cout << s.back();
			}
			std::cout << "\n";
		}
	}
};

int main() {
	std::ios::sync_with_stdio(0);
	std::cin.tie(0);
	Solution p;
	int m;
	std::cin >> m;
	while (m--) {
		p.solve();
	}
	return 0;
}
